src/render/vulkan/SDL_render_vulkan.c

```diff
@@ -3720,7 +3720,7 @@ static int VULKAN_RunCommandQueue(SDL_Renderer *renderer, SDL_RenderCommand *cmd
         rendererData->recreateSwapchain = SDL_FALSE;
     }
 
-    if (VULKAN_UpdateVertexBuffer(renderer, vertices, vertsize, &stateCache) < 0) {
+    if (!VULKAN_UpdateVertexBuffer(renderer, vertices, vertsize, &stateCache)) {
```

Comment:
Thinking about it, this one is weird? Is `dataSizeInBytes == 0` considered an error case or, as the note says in the procedure, a `nothing to do` case which should be silently ignored? I.e., should we change `VULKAN_UpdateVertexBuffer` to return int instead of bool? If not, we should change the `0` return to `SDL_FALSE` for better readability, and also set an SDL_error?

Reply:
I agree this is a good change, I'll fold it in with the return value changes I'm making.